File: app/src/main/java/ru/samtakoy/core/services/import_utils/FromUriStreamFactory.java
package ru.samtakoy.core.services.import_utils;

import android.content.ContentResolver;
import android.net.Uri;

import java.io.InputStream;

import ru.samtakoy.core.database.UriUtils;

public class FromUriStreamFactory implements StreamFactory{


    private ContentResolver mResolver;
    private Long mTargetThemeId;
    private Uri mSelectedFileUri;

    public FromUriStreamFactory(
            ContentResolver resolver, Long targetThemeId, Uri selectedFileUri
    ){

        mResolver = resolver;
        mTargetThemeId = targetThemeId;
        mSelectedFileUri = selectedFileUri;
    }

    @Override
    public InputStream openStream() throws Exception {
        return mResolver.openInputStream(mSelectedFileUri);
    }

    @Override
    public String getSrcPath() {
        return mSelectedFileUri.getPath();
    }

    @Override
    public Long getThemeId() {
        return mTargetThemeId;
    }

    @Override
    public String getFileName() {
        return UriUtils.getFileNameByUri(mResolver, mSelectedFileUri);
    }

    @Override
    public ContentResolver getResolver() {
        return mResolver;
    }
}

File: app/src/main/kotlin/ru/samtakoy/features/import_export/utils/FromUriStreamFactory.kt
package ru.samtakoy.features.import_export.utils

import android.content.ContentResolver
import android.net.Uri
import ru.samtakoy.features.import_export.utils.UriUtils.getFileNameByUri
import java.io.InputStream

class FromUriStreamFactory(
    private val mResolver: ContentResolver,
    override val themeId: Long,
    private val mSelectedFileUri: Uri
) : StreamFactory {
    @Throws(Exception::class) override fun openStream(): InputStream {
        return mResolver.openInputStream(mSelectedFileUri)!!
    }

    override val srcPath: String
        get() = mSelectedFileUri.getPath()!!

    override val fileName: String
        get() = getFileNameByUri(mResolver, mSelectedFileUri)
}
